src/plugins/data/server/search/strategies/esql_async_search/esql_async_search_strategy.ts

```diff
@@ -0,0 +1,158 @@
+import { catchError, tap } from 'rxjs/operators';
+import { getKbnServerError } from '@kbn/kibana-utils-plugin/server';
+import { SqlQueryRequest } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { SqlGetAsyncResponse } from '@elastic/elasticsearch/lib/api/types';
+import { getKbnSearchError } from '../../report_search_error';
+import type { ISearchStrategy, SearchStrategyDependencies } from '../../types';
+import type { IAsyncSearchOptions } from '../../../../common';
+import { IKibanaSearchRequest, IKibanaSearchResponse, pollSearch } from '../../../../common';
+import { getDefaultAsyncGetParams, getDefaultAsyncSubmitParams } from './request_utils';
+import { toAsyncKibanaSearchResponse } from './response_utils';
+import { SearchConfigSchema } from '../../../../config';
+
+export const esqlAsyncSearchStrategyProvider = (
+  searchConfig: SearchConfigSchema,
+  logger: Logger,
+  useInternalUser: boolean = false
+): ISearchStrategy<
+  IKibanaSearchRequest<SqlQueryRequest['body']>,
+  IKibanaSearchResponse<SqlGetAsyncResponse>
+> => {
+  function cancelAsyncSearch(id: string, esClient: IScopedClusterClient) {
+    const client = useInternalUser ? esClient.asInternalUser : esClient.asCurrentUser;
+    return client.transport.request(
+      {
+        method: 'DELETE',
+        path: `/_query/async/${id}`,
+      },
+      {
+        meta: true,
+        // we don't want the ES client to retry (default value is 3)
+        maxRetries: 0,
+      }
+    );
+  }
+
+  function asyncSearch(
+    { id, ...request }: IKibanaSearchRequest<SqlQueryRequest['body']>,
+    options: IAsyncSearchOptions,
+    { esClient, uiSettingsClient }: SearchStrategyDependencies
+  ) {
+    const client = useInternalUser ? esClient.asInternalUser : esClient.asCurrentUser;
+
+    const search = async () => {
+      const params = id
+        ? {
+            ...getDefaultAsyncGetParams(searchConfig, options),
+            ...(request.params?.keep_alive ? { keep_alive: request.params.keep_alive } : {}),
+            ...(request.params?.wait_for_completion_timeout
+              ? { wait_for_completion_timeout: request.params.wait_for_completion_timeout }
+              : {}),
+          }
+        : {
+            ...(await getDefaultAsyncSubmitParams(uiSettingsClient, searchConfig, options)),
+            ...request.params,
+          };
+      const { body, headers, meta } = id
+        ? await client.transport.request(
+            { method: 'GET', path: `/_query/async/${id}`, querystring: { ...params } },
+            { ...options.transport, signal: options.abortSignal, meta: true }
+          )
+        : await client.transport.request(
+            { method: 'POST', path: `/_query/async`, body: params },
+            { ...options.transport, signal: options.abortSignal, meta: true }
+          );
+
+      const finalResponse = toAsyncKibanaSearchResponse(
+        body as SqlGetAsyncResponse,
```

Comment:
I think you can avoid the casting here by passing the types to the `client.transport.request` call above:

```ts
await client.transport.request<SqlGetAsyncResponse>(
            { method: 'POST', path: `/_query/async`, body: params },
            { ...options.transport, signal: options.abortSignal, meta: true }
          );
```